.github/workflows/build-release.yml

```diff
@@ -70,3 +70,33 @@ jobs:
 
               run: |
                   gh pr comment $PR_NUMBER --body "$COMMENT_BODY" --repo "$GITHUB_REPOSITORY"
+
+    tag:
+        name: Create release tag
+        runs-on: ubuntu-latest
+        needs: build
+
+        steps:
+            - name: Create GitHub App Token
+              uses: actions/create-github-app-token@v1
+              id: app-token
+              with:
+                  app-id: ${{ vars.RELEASE_BOT_APP_ID }}
+                  private-key: ${{ secrets.RELEASE_BOT_PRIVATE_KEY }}
+
+            - name: Checkout repository
+              uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.event.pull_request.base.sha }}
```

Comment:
a little unsure about if this is the correct value. the intention for this step is to check out the relevant `release` branch that the PR got merged into

Reply:
I believe the default for this action is what we want

Reply:
it's never clear to me when branch/ref is checked out by default. the implicit relationship between the workflow event and this action always confuses me 😂

Reply:
addressed via e2e0fdf

Reply:
I think it actually runs of the "imaginary" merge commit of merging the PR.

Reply:
See https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request

> `GITHUB_REF` = PR merge branch refs/pull/PULL_REQUEST_NUMBER/merge

And:

> The value of GITHUB_REF varies for a closed pull request depending on whether the pull request has been merged or not. If a pull request was closed but not merged, it will be refs/pull/PULL_REQUEST_NUMBER/merge. If a pull request was closed as a result of being merged, it will be the fully qualified ref of the branch it was merged into, for example /refs/heads/main.

The checkout action uses GITHUB_REF by default: https://github.com/actions/checkout

Reply:
based on that, seems like omitting it is fine right?